File: csv/src/main/scala/akka/stream/alpakka/csv/scaladsl/CsvFraming.scala
/*
 * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
 */
package akka.stream.alpakka.csv.scaladsl

import akka.NotUsed
import akka.stream.alpakka.csv.CsvFramingStage
import akka.stream.scaladsl.Flow
import akka.util.ByteString

object CsvFraming {

  val Backslash: Byte = '\\'
  val Comma: Byte = ','
  val SemiColon: Byte = ';'
  val Colon: Byte = ':'
  val Tab: Byte = '\t'
  val DoubleQuote: Byte = '"'

  /** Creates CSV framing flow that separates CSV lines from incoming
   * [[akka.util.ByteString]] objects.
   */
  def lineScanner(delimiter: Byte = Comma,
                  quoteChar: Byte = DoubleQuote,
                  escapeChar: Byte = Backslash): Flow[ByteString, List[ByteString], NotUsed] =
    Flow.fromGraph(new CsvFramingStage(delimiter, quoteChar, escapeChar))
}

File: csv/src/main/java/akka/stream/alpakka/csv/javadsl/CsvFraming.java
package akka.stream.alpakka.csv.javadsl;

import akka.NotUsed;
import akka.stream.javadsl.Flow;
import akka.util.ByteString;
import scala.collection.JavaConversions;

import java.util.Collection;

public class CsvFraming {

    public static final byte backslash = '\\';
    public static final byte comma = ',';
    public static final byte doubleQuote = '"';

    public static Flow<ByteString, Collection<ByteString>, NotUsed> lineScanner() {
        return lineScanner(comma, doubleQuote, backslash);
    }

    public static Flow<ByteString, Collection<ByteString>, NotUsed> lineScanner(byte delimiter, byte quoteChar, byte escapeChar) {
        return akka.stream.alpakka.csv.scaladsl.CsvFraming
                .lineScanner(delimiter, quoteChar, escapeChar).asJava()
                .map(JavaConversions::asJavaCollection);
    }


}
